posthog/tasks/process_scheduled_changes.py

```diff
@@ -0,0 +1,35 @@
+from posthog.models import ScheduledChange
+from django.utils import timezone
+from posthog.models import FeatureFlag
+from django.db import transaction, OperationalError
+
+models = {"FeatureFlag": FeatureFlag}
+
+
+def process_scheduled_changes() -> None:
+    try:
+        with transaction.atomic():
+            scheduled_changes = ScheduledChange.objects.select_for_update(nowait=True).filter(
+                executed_at__isnull=True,
+                scheduled_at__lt=timezone.now(),
+            )
+
+            for scheduled_change in scheduled_changes.iterator(chunk_size=10000):
+                try:
+                    # Execute the change on the model instance
+                    model = models[scheduled_change.model_name]
+                    instance = model.objects.get(id=scheduled_change.record_id)
+                    instance.scheduled_changes_dispatcher(scheduled_change.payload)
+
+                    # Mark scheduled change completed
+                    scheduled_change.executed_at = timezone.now()
+                    scheduled_change.save()
+
+                except Exception as e:
+                    # Store the failure reason
+                    scheduled_change.failure_reason = str(e)
+                    scheduled_change.executed_at = timezone.now()
+                    scheduled_change.save()
+    except OperationalError:
+        # Failed to obtain the lock
```

Comment:
should this be an explicitly raised error?

I'm thinking about how I'd investigate this if I thought it was broken and you were off... 

I suppose regardless I could look at the generic emitted scheduled, succeeded, and failed counts for this task

And since they are Prometheus metrics, they also give you a mechanism to alert if you want to...

So not-blocking. 

You can copy http://grafana-prod-eu/d/celery-workers/celery-workers?orgId=1 if you want to set a dashboard specifically for the feature

Reply:
No I don't think so, this will be intended behaviour (when the lock isn't acquired that is), but everything else is unknown indeed and should raise for easy debugging.

Because multiple workers will compete for this lock to process changes